### PR TITLE
39 nifti writer

### PR DESCRIPTION
**Pull request by Nic-Ma**
Hi All,

This PR implemented the tool API to write model output data into **Nifti format** image #39 .
We can call it when evaluation and validation examples are ready.
Could you please help review it?
Thanks.

### CODE REVIEW

**Comment by Nic-Ma** [comment]
Hi @IsaacYangSLA ,

I found 2 CI issues in this PR:
1. I can't open the ci/gitlab/gitlab.com web page for details, always loading:
image
   Some China firewall issue with IT? Just like what we faced before..
2. @wyli told me that this error is related to installation failure of PyTorch 1.4, seems not related to PR?

Could you please help take a look?
Thanks.

File: monai/data/writers/niftiwriter.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        revert_canonical (bool): whether to revert canonical.
+        dtype (np.dtype, optional): convert the loaded image to this data type.
+
+    """
+    assert isinstance(data, np.ndarray), 'input data must be numpy array.'
+    if affine is None:
+        affine = np.eye(4)
+
+    if revert_canonical:
+        codes = nib.orientations.axcodes2ornt(nib.orientations.aff2axcodes(np.linalg.inv(affine)))
```
Comment (ericspod) [inline monai/data/writers/niftiwriter.py]: Does `np.linalg.inv(affine)` actually give up the orientation we expect of the original? If the original's axis codes were LAS they would become RAS when converted to canonical, but doesn't this line turn RAS into LPI?

Comment (Nic-Ma) [reply monai/data/writers/niftiwriter.py]: Hi @wyli ,

As you are expert at Nifti format, could you please help take a look at @ericspod 's question?
Thanks.

**Comment by wyli** [comment]
all commits from this PR are merged into #56 please review #56 instead

File: monai/data/writers/niftiwriter.py (hunk 1 above)
Comment (madil90) [reply monai/data/writers/niftiwriter.py]: @ericspod  #74 fixes the canonical reversion logic. We need to store the original affine matrix, use it to get orientation and apply the orientation to revert data.

**Pull request closed without merging**